src/model_metric_reporter.cpp

```diff
@@ -28,6 +29,11 @@ constexpr int NUMBER_OF_BUCKETS = 33;
 constexpr double BUCKET_POWER_BASE = 1.8;
 constexpr double BUCKET_MULTIPLIER = 10;
 
+#define THROW_IF_NULL(VAR, MESSAGE)      \
+    if (VAR == nullptr) {                \
```

Comment:
Either add spdlog here & verify if it will correctly pass __LINE__ info or you should differentiate message for easier debug in case of errors.

Reply:
Added spdlog